### PR TITLE
fix(crash): Fix game crash on short error messages

### PR DESCRIPTION
**Pull request by tibetiroka**
**Bug fix**

This PR fixes a crash reported by @Arachi-Lover on discord.

## Summary
The game crashed when loading a save file. No longer crashes after this PR.

## Testing Done
I tested that the game no longer crashes when the save is loaded.

## Save File
[Arachi's save file](https://discord.com/channels/251118043411775489/536900466655887360/1211341115165380618)

## Performance Impact
N/A

### CODE REVIEW

File: source/main.cpp [hunk 1]
```diff
@@ -95,7 +95,8 @@ int main(int argc, char *argv[])
 	// Ensure that we log errors to the errors.txt file.
 	Logger::SetLogErrorCallback([&hasErrors](const string &errorMessage) {
 		static const string PARSING_PREFIX = "Parsing: ";
-		if(errorMessage.substr(PARSING_PREFIX.length()) != PARSING_PREFIX)
+		if(errorMessage.length() <= PARSING_PREFIX.length() || errorMessage.substr(PARSING_PREFIX.length())
+				!= PARSING_PREFIX)
```
Comment (RisingLeaf) [inline source/main.cpp:99]:
```suggestion
		if(errorMessage.substr(std::min(PARSING_PREFIX.length(), errorMessage.length()))
				!= PARSING_PREFIX)
```
what about this?

Comment (RisingLeaf) [reply source/main.cpp:99]: I guess it is the same

Comment (tibetiroka) [reply source/main.cpp:99]: Honestly, I don't know which one is better... if we go with your solution, do we still need the line break?

Comment (RisingLeaf) [reply source/main.cpp:99]: but we less than 120

Comment (tibetiroka) [reply source/main.cpp:99]: It's going in, then.

Comment (tibetiroka) [reply source/main.cpp:99]:
```suggestion
		if(errorMessage.substr(min(PARSING_PREFIX.length(), errorMessage.length())) != PARSING_PREFIX)
```

**Review by RisingLeaf — APPROVED** [review]
worx

**Review by OcelotWalrus — APPROVED** [review]

File: source/main.cpp [hunk 2]
```diff
@@ -95,7 +95,7 @@ int main(int argc, char *argv[])
 	// Ensure that we log errors to the errors.txt file.
 	Logger::SetLogErrorCallback([&hasErrors](const string &errorMessage) {
 		static const string PARSING_PREFIX = "Parsing: ";
-		if(errorMessage.substr(PARSING_PREFIX.length()) != PARSING_PREFIX)
+		if(errorMessage.substr(min(PARSING_PREFIX.length(), errorMessage.length())) != PARSING_PREFIX)
```
Comment (warp-core) [inline source/main.cpp:98]: The thing was wrong to begin with. It should be:
```suggestion
		if(errorMessage.substr(0, PARSING_PREFIX.length()) != PARSING_PREFIX)
```
No exception is thrown if `pos + count > size()` so no additional bounds checking is necessary.

https://en.cppreference.com/w/cpp/string/basic_string/substr

Comment (tibetiroka) [reply source/main.cpp:98]: https://github.com/endless-sky/endless-sky/commit/9215f8f7029ecfa84ac8c9747cda021b92e1c28e

**Comment by tibetiroka** [comment]
The CI is failing because of a network error. Other stuff seems to pass.

**Pull request merged**